duckduckgo/Android
# Add internal features

Task/Issue URL: https://app.asana.com/0/414730916066338/1200142941651446/f
Tech Design URL: 
CC: 

**Description**:
This PR adds some intrastructure to easily create `internal` features what will have an entry point in the settings screen. Along with the first feature to be able to test new TDS lists before they hit production.

**How to add a new `internal` feature?**
The process to add a new entry point to an internal feature is easy:
* Create a class that extends `InternalFeaturePlugin` and return it as a multibinding

```kotlin
@ContributesMultibinding(AppObjectGraph::class)
class MyInternalFeature @Inject constructor() : InternalFeaturePlugin {
    override fun internalFeatureTitle(): String {
        return "My Internal feature"
    }

    override fun internalFeatureSubtitle(): String {
        return "brief summary of the feature"
    }

    override fun onInternalFeatureClicked(activityContext: Context) {
        context.startActivity(MyInternalFeatureActivity.intent(context))
    }
}
```
* `internalFeatureTitle` returns the title of the internal feature that will appear in the settings screen
* `internalFeatureSubtitle` returns summary info about the internal feature that will appear in the settings screen
* `onInternalFeatureClicked` will be called when the user clicks on the internal feature. This is a good place to launch the UX for the internal feature, eg. and `Activity`



**Steps to test this PR**:
1. install `internalDebug` or `internalRelease` product flavors from this branch
1. go to settings and scroll all the way to the bottom
1. verify there is a new section called `Internal features`
1. verify that section contains a `Developer settings` entry
1. install any other product flavor (`play` or `fdroid`)
1. go to settings and scroll all the way to the bottom
1. verify the `Internal features` do not appear


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
